src/task-dispatcher/td.hh

```diff
@@ -77,14 +87,21 @@ public:
 
     future(future&) = delete;
     future& operator=(future&) = delete;
-    future(future&&) noexcept = default;
-    future& operator=(future&&) noexcept = default;
+
+    future(future&& rhs) noexcept : mSync(rhs.mSync), mValue(std::move(rhs.mValue)) { rhs.mSync.initialized = false; }
+    future& operator=(future&& rhs) noexcept
+    {
+        mSync = rhs.mSync;
+        mValue = std::move(rhs.mValue);

```

Comment:
`mValue` is `const` and thus cannot be moved at all. `const` members are usually not recommended because they pessimize moves